mongo/errors.go

```diff
@@ -52,10 +52,12 @@ func replaceErrors(err error) error {
 		return nil
 	}
 
-	if err == topology.ErrTopologyClosed {
+	if errors.Is(err, topology.ErrTopologyClosed) {
 		return ErrClientDisconnected
 	}
-	if de, ok := err.(driver.Error); ok {
+
+	var de driver.Error
+	if errors.As(err, &de) {
```

Comment:
This comparison is not the same as `de, ok := err.(driver.Error); ok` in that *driver.Error does not implement the [error](https://pkg.go.dev/builtin#error) interface. I'm realizing that this pattern of using `errors.As` with a to compare to a target who's sole purpose is the comparison. I think we should avoid doing this and retain the existing type assertion.

Would you mind revert all code with this pattern:

```go
var target Error
if errors.As(err, &target) {}
```

Back to using a type assertion?

Reply:
Thanks @prestonvasquez for the review and sharing feedback!

Updated PR.